ambry-protocol/src/main/java/com.github.ambry.protocol/DeleteRequest.java

```diff
@@ -84,7 +121,103 @@ public String toString() {
     sb.append("BlobID=").append(blobId);
     sb.append(", ").append("ClientId=").append(clientId);
     sb.append(", ").append("CorrelationId=").append(correlationId);
+    sb.append(", ").append("AccountId=").append(accountId);
+    sb.append(", ").append("ContainerId=").append(containerId);
+    sb.append(", ").append("DeletionTimeInSecs=").append(deletionTimeInSecs);
     sb.append("]");
     return sb.toString();
   }
+
+  /**
+   * Class to read protocol version 1 DeleteRequest from the stream.
+   */
+  private static class DeleteRequest_V1 {
+    static ReceivedDeleteRequest readFrom(DataInputStream stream, ClusterMap map) throws IOException {
+      int correlationId = stream.readInt();
+      String clientId = Utils.readIntString(stream);
+      BlobId id = new BlobId(stream, map);
+      return new ReceivedDeleteRequest(correlationId, clientId, id);
+    }
+  }
+
+  /**
+   * Class to read protocol version 2 DeleteRequest from the stream.
+   */
+  private static class DeleteRequest_V2 {
+    static ReceivedDeleteRequest readFrom(DataInputStream stream, ClusterMap map) throws IOException {
+      int correlationId = stream.readInt();
+      String clientId = Utils.readIntString(stream);
+      BlobId id = new BlobId(stream, map);
+      short accountId = stream.readShort();
+      short containerId = stream.readShort();
+      int deletionTimeInSecs = stream.readInt();
+      return new ReceivedDeleteRequest(correlationId, clientId, id, accountId, containerId, deletionTimeInSecs);
+    }
+  }
+
+  /**
+   * Class that represents a DeleteRequest that was received and cannot be sent out.
+   */
+  public static class ReceivedDeleteRequest {
```

Comment:
What is the necessity for this class? `PutRequest` has a `ReceivedPutRequest` because they differ a lot in the fields that they have - I don't see any differences b/w `DeleteRequest` and `ReceivedDeleteRequest`